### PR TITLE
add NSFW field to RequestWiki T12724

### PR DESCRIPTION
**Pull request by Reception123**

### CODE REVIEW

**Review by Universal-Omega — APPROVED** [review]

**Comment by Reception123** [comment]
Currently blocked by:
`
[b6894481196641c0f1faf443] /wiki/Special:RequestWikiQueue/32 DomainException: MediaWiki\HTMLForm\HTMLForm::getField: no field named nsfw

Backtrace:

from /srv/mediawiki/1.43/includes/htmlform/HTMLForm.php(539)
#0 /srv/mediawiki/1.43/includes/htmlform/HTMLFormField.php(194): MediaWiki\HTMLForm\HTMLForm->getField(string)
#1 /srv/mediawiki/1.43/includes/htmlform/HTMLFormField.php(209): MediaWiki\HTMLForm\HTMLFormField->getNearestField(string, bool)
#2 /srv/mediawiki/1.43/includes/htmlform/HTMLFormField.php(326): MediaWiki\HTMLForm\HTMLFormField->getNearestFieldValue(array, string, bool, bool)
#3 /srv/mediawiki/1.43/includes/htmlform/HTMLFormField.php(392): MediaWiki\HTMLForm\HTMLFormField->checkStateRecurse(array, array)
#4 /srv/mediawiki/1.43/includes/htmlform/HTMLFormField.php(405): MediaWiki\HTMLForm\HTMLFormField->isHidden(array)
#5 /srv/mediawiki/1.43/includes/htmlform/HTMLForm.php(2192): MediaWiki\HTMLForm\HTMLFormField->isDisabled(array)
#6 /srv/mediawiki/1.43/includes/htmlform/HTMLForm.php(664): MediaWiki\HTMLForm\HTMLForm->loadFieldData()
#7 /srv/mediawiki/1.43/includes/htmlform/HTMLForm.php(716): MediaWiki\HTMLForm\HTMLForm->prepareForm()
#8 /srv/mediawiki/1.43/extensions/CreateWiki/includes/RequestWiki/SpecialRequestWikiQueue.php(140): MediaWiki\HTMLForm\HTMLForm->show()
#9 /srv/mediawiki/1.43/extensions/CreateWiki/includes/RequestWiki/SpecialRequestWikiQueue.php(56): Miraheze\CreateWiki\RequestWiki\SpecialRequestWikiQueue->lookupRequest(string)
#10 /srv/mediawiki/1.43/includes/specialpage/SpecialPage.php(723): Miraheze\CreateWiki\RequestWiki\SpecialRequestWikiQueue->execute(string)
#11 /srv/mediawiki/1.43/includes/specialpage/SpecialPageFactory.php(1724): MediaWiki\SpecialPage\SpecialPage->run(string)
#12 /srv/mediawiki/1.43/includes/actions/ActionEntryPoint.php(504): MediaWiki\SpecialPage\SpecialPageFactory->executePath(string, MediaWiki\Context\RequestContext)
#13 /srv/mediawiki/1.43/includes/actions/ActionEntryPoint.php(146): MediaWiki\Actions\ActionEntryPoint->performRequest()
#14 /srv/mediawiki/1.43/includes/MediaWikiEntryPoint.php(200): MediaWiki\Actions\ActionEntryPoint->execute()
#15 /srv/mediawiki/config/initialise/entrypoints/index.php(98): MediaWiki\MediaWikiEntryPoint->run()
#16 {main}`

File: includes/Hooks.php [hunk 1]
```diff
@@ -141,6 +147,67 @@ public static function factory(
...
+			$formDescriptor,
+			afterKey: 'edit-bio',
+			newKey: 'edit-nsfw',
+			newField: $nsfwField
+		);
+		$nsfwtextField = [
+			'label-message' => 'requestwiki-label-nsfwtext',
+			'type' => 'text',
+			'section' => 'editing',
+			'hide-if' => [ '!==', 'nsfw', '1' ],
```
Comment (Universal-Omega) [inline includes/Hooks.php:199]: 'hide-if' => [ '!==', 'edit-nsfw', '1' ],

File: includes/Hooks.php [hunk 2]
```diff
@@ -141,6 +147,67 @@ public static function factory(
...
+			'default' => $wikiRequestManager->getExtraFieldData( 'nsfw' ),
+		];
+
+		RequestWikiFormUtils::insertFieldAfter(
+			$formDescriptor,
+			afterKey: 'edit-bio',
+			newKey: 'edit-nsfw',
+			newField: $nsfwField
+		);
+		$nsfwtextField = [
```
Comment (BlankEclair) [inline includes/Hooks.php:195]: Newline?

File: includes/Hooks.php [hunk 3]
```diff
@@ -141,6 +147,67 @@ public static function factory(
...
+			'type' => 'check',
+		];
+
+		RequestWikiFormUtils::insertFieldAfter(
+			$formDescriptor,
+			afterKey: 'bio',
+			newKey: 'nsfw',
+			newField: $nsfwField
+		);
+		$nsfwtextField = [
```
Comment (BlankEclair) [inline includes/Hooks.php:163]: Newline?

**Pull request merged**